modules/fluid_properties/src/fluidproperties/TabulatedFluidProperties.C

```diff
@@ -895,6 +918,100 @@ TabulatedFluidProperties::vaporPressure(Real temperature, Real & psat, Real & dp
     mooseError("vaporPressure not specified.");
 }
 
+Real
+TabulatedFluidProperties::vaporTemperature(Real pressure) const
+{
+
+  const FPDualReal p = pressure;
+
+  return vaporTemperature_ad(p).value();
+}
+
+void
+TabulatedFluidProperties::vaporTemperature(Real pressure, Real & Tsat, Real & dTsat_dp) const
+{
+
+  if (_fp)
+    _fp->vaporTemperature(pressure, Tsat, dTsat_dp);
+  else
+    mooseError("vaporTemperature not specified.");
```

Comment:
It's not really that it s not specified. It s more:

"Tabulation does not contain the vapor temperature. To use the TabulatedFluidProperties and compute the vaporTemperature, please provide another fluid property using the 'fp' parameter, which will be used to compute the vapor temperature"

and same, make it a standard error message in a TFP routine, which you can re-use for each of these routines